src/VisualStudio/Core/Def/Implementation/FindReferences/Entries/AbstractDocumentSpanEntry.cs

```diff
@@ -86,6 +90,32 @@ public override bool TryCreateColumnContent(string columnName, out FrameworkElem
             }
 
             protected abstract IList<Inline> CreateLineTextInlines();
+
+            public static async Task<MappedSpanResult?> TryMapAndGetFirstAsync(DocumentSpan documentSpan, SourceText sourceText, CancellationToken cancellationToken)
+            {
+                var service = documentSpan.Document.Services.GetService<ISpanMappingService>();
+                if (service == null)
+                {
+                    return new MappedSpanResult(documentSpan.Document.FilePath, sourceText.Lines.GetLinePositionSpan(documentSpan.SourceSpan), documentSpan.SourceSpan);
+                }
+
+                var results = await service.MapSpansAsync(
+                    documentSpan.Document, SpecializedCollections.SingletonEnumerable(documentSpan.SourceSpan), cancellationToken).ConfigureAwait(false);
+
+                if (results.IsDefaultOrEmpty)
+                {
+                    return new MappedSpanResult(documentSpan.Document.FilePath, sourceText.Lines.GetLinePositionSpan(documentSpan.SourceSpan), documentSpan.SourceSpan);
+                }
+
+                return results.FirstOrNullable(r => !r.IsDefault);
+            }
+
+            public static SourceText GetLineContainingPosition(SourceText text, int position)
+            {
+                var line = text.Lines.GetLineFromPosition(position);
+
+                return text.GetSubText(line.Span);
```

Comment:
I prefer this way.

Reply:
my point is that it seems weird that you then need to know that these entries are *not* related to each other.  When you have SourceText+Span, it is saying "this is the span that is valid within me".  But here we have a different formalization.  That is not how we do things elsewhere in the IDE code and it can definitley make things more confusing. For example, someone might assume (thanks to how we do things everywhere else) that you can just index safely into this text with teh span positions.  But that doesn't seem to be the case.

In general, the most consistent code is with teh rest of the IDE, the safer and easier it is for people to work with it.  You can use the same patterns and practices used elsewhere, and don't have ot be super cautious that things are being done in a different fashion in this one particular case.